CAIPs/caip-222.md

```diff
@@ -0,0 +1,249 @@
+---
+caip: 222
+title: Wallet Authenticate JSON-RPC Method
+author: Pedro Gomes (@pedrouid)
+discussions-to: https://github.com/ChainAgnostic/CAIPs/discussions/222
+status: Draft
+type: Standard
+created: 2022-04-07
+requires: ["2", "10", "74", "122"]
+---
+
+## Simple Summary
+
+This CAIP defines a JSON-RPC method to authenticate wallet ownership/control of a blockchain account.
+
+## Abstract
+
+This proposal provides a simpler user experience by bundling two wallet requests into a single method to provide wallet authentication when exposing blockchain accounts with a cryptographic signature
+
+## Motivation
+
+Prior this standard Wallets were required to establish a JSON-RPC connection with an application (Dapp) to expose blockchain accounts but this required a secondary request to verify ownership of each account to return a cryptographic signature.
+
+There are challenges regarding the flexibility of what can be signed or not without prior knowledge of the blockchain accounts but with the adoption of standards like SIWx ([CAIP-122][caip-122]) we have the ability to define specific parameters that can be signed securely without the accounts being exposed beforehand.
+
+This will accelerate wallet adoption by social media applications which do not require persistent sessions and only need a single cryptographic signature to verify ownership of blockchain accounts.
+
+## Specification
+
+This JSON-RPC method can be requested to a wallet provider without prior knowledge of the blockchain accounts.
+
+The requester will provide parameters required by [CAIP-122][caip-122] plus a CACAO header type as specified by [CAIP-74][caip-74]
+
+The responder will return a signed CACAO with a header type and payload matching the requested parameters
+
+### Request
+
+The application would interface with a provider to make request as follows:
+
+```jsonc
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "method": "wallet_authenticate",
+  "params": {
+      "cacaov": string,
+      "type": string,
+      "chainId": string,
+      "domain": string,
+      "aud": string,
+      "version": string,
+      "nonce": string,
+      "iat": string,
+      "exp": string, // optional
+      "nbf": string, // optional
+      "statement": string, // optional
+      "requestId": string, // optional
+      "resources": string[], // optional
+  }
+}
+```
+
+The JSON-RPC method is labelled as `wallet_authenticate` and expects the following parameters:
+
+- cacaov - Cacao version number
+- type - Cacao header message type
+- chainId - [CAIP-2][]-defined `chainId` to identify specific chain or network.
+- domain - [RFC 4501][rfc 4501] `dnsauthority` that is requesting the signing.
+- aud - [RFC 3986][rfc 3986] URI referring to the resource that is the subject of the signing.
+- version - Current version of the message.
+- nonce - Randomized token to prevent signature replay attacks.
```

Comment:
We should specify the requirements of the nonce here too.

Reply:
Wouldn't that be part of CAIP-122 scope instead?

Reply:
If we reference directly to 122 as the normative definition then yeah that makes sense. I think that's actually the best way to do this too. I can help double check it to make sure we got this normatively defined well.